game/src/game/services/GameStateService.cpp

```diff
@@ -505,6 +623,116 @@ void GameStateService::PlayerCompletedLap(PlayerRecord& player)
+
+            for (size_t i = 0; i < race_state_.sorted_players.size(); i++)
+            {
+                const int place = static_cast<int>(i + 1);
+                Entity* entity = race_state_.sorted_players[i]->entity;
+
+                ImGui::PushID(entity->GetId());
+                ImGui::PushFont(font_cookie_);
+                if (place == 1 && race_state_.sorted_players[i]->is_human)
+                    ImGui::Text("%d\t%s\t", place, entity->GetName().c_str());
+                else if (place == 1 && !race_state_.sorted_players[i]->is_human)
+                    ImGui::Text("%d\t%s\t\t", place, entity->GetName().c_str());
+                else if (place == 2 && race_state_.sorted_players[i]->is_human)
+                    ImGui::Text("%d\t%s\t", place, entity->GetName().c_str());
+                else if (place == 2 && !race_state_.sorted_players[i]->is_human)
+                    ImGui::Text("%d\t%s\t\t", place, entity->GetName().c_str());
+                else if (place == 3 && race_state_.sorted_players[i]->is_human)
+                    ImGui::Text("%d\t%s\t", place, entity->GetName().c_str());
+                else if (place == 3 && !race_state_.sorted_players[i]->is_human)
+                    ImGui::Text("%d\t%s\t\t", place, entity->GetName().c_str());
+                else if (place == 4 && race_state_.sorted_players[i]->is_human)
+                    ImGui::Text("%d\t%s\t", place, entity->GetName().c_str());
+                else if (place == 4 && !race_state_.sorted_players[i]->is_human)
+                    ImGui::Text("%d\t%s\t\t", place, entity->GetName().c_str());
+
+                ImGui::SameLine(0.f, 100.f);
+                int min = (int)player.finished_time / 60;
+                int second = (int)player.finished_time % 60;
+                ImGui::Text("%02d:%02d:%02.0f", min, second,
+                            (player.finished_time - (min * 60 + second)) * 100);
+                ImGui::PopFont();
+                ImGui::PopID();
+            }
+            ImGui::End();
+
+            ImGui::SetNextWindowPos(ImVec2(ImGui::GetIO().DisplaySize.x - 80,
+                                           ImGui::GetIO().DisplaySize.y - 80));
+            ImGui::Begin("home", nullptr, flags);
+            ImGui::PushStyleVar(ImGuiStyleVar_FrameRounding, 50.0f);
+
+            ImGui::PushStyleColor(ImGuiCol_Button, ImVec4(0.f, 0.f, 0.f, 0.f));
+            ImGui::PushStyleColor(ImGuiCol_ButtonActive,
+                                  ImVec4(0.f, 0.f, 0.f, 0.f));
+            ImGui::PushStyleColor(ImGuiCol_ButtonHovered,
+                                  ImVec4(1.f, 1.f, 1.f, 0.1f));
+            if (ImGui::ImageButton("home button", home_button_->GetGuiHandle(),
+                                   ImVec2(40, 37)))
+            {
+                scene_service_->SetActiveScene("MainMenu");
+            }
+            ImGui::PopStyleColor(3);
+            ImGui::PopStyleVar(1);
+
+            ImGui::End();
```

Comment:
I think this won't display on the screen at all. Anything that has to do with ImGui should be inside of the `OnGui` functions, otherwise, it might break, do weird things, or just do nothing.

Instead, when the race is over, you should set `race_state_.state = GameState::kPostRace`, and then in the `ImGui` function, show the scoreboard if we're in that state